redash/query_runner/__init__.py

```diff
@@ -67,6 +67,22 @@ def configuration_schema(cls):
     def run_query(self, query):
         raise NotImplementedError()
 
+    def fetch_columns(self, columns):
+        column_names = []
+        duplicates_counter = 1
+        new_columns = []
+
+        for col in columns:
+            column_name = col[0]
+            if column_name in column_names:
+                column_name += " " + str(duplicates_counter)
```

Comment:
Also you can use string formatting: `column_name = "{}_{}".format(column_name, duplicates_counter)`.